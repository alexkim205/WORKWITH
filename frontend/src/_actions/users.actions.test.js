import createMockStore from '../_config/mockStore.config';
import {
  getUsers,
  getUser,
  login,
  register,
  logout,
  updateUser,
  deleteUser
} from './users.actions';
import testConstants from '../_constants/test.constants';

const {
  users: { user, loginCredentials, newUser, newUpdateUser }
} = testConstants;

describe('User Actions', () => {
  let store;
  beforeAll(() => {
    store = createMockStore();
  });
  afterEach(() => {
    store.clearActions();
  });
  it('it should GET users', async () => {
    await store.dispatch(getUsers());
    expect(store.getActions()).toMatchSnapshot();
  });
  it('it should GET a user', async () => {
    await store.dispatch(getUser(user._id));
    expect(store.getActions()).toMatchSnapshot();
  });
  it('it should LOGIN a user', async () => {
    await store.dispatch(
      login(loginCredentials.email, loginCredentials.password)
    );
    expect(store.getActions()).toMatchSnapshot();
  });
  it('it should REGISTER a user', async () => {
    await store.dispatch(register(newUser));
    expect(store.getActions()).toMatchSnapshot();
  });
  it('it should LOGOUT a user', async () => {
    await store.dispatch(logout());
    expect(store.getActions()).toMatchSnapshot();
  });
  it('it should UPDATE a user', async () => {
    await store.dispatch(updateUser(user._id, newUpdateUser));
    expect(store.getActions()).toMatchSnapshot();
  });
  it('it should DELETE a note', async () => {
    await store.dispatch(deleteUser(user._id));
    expect(store.getActions()).toMatchSnapshot();
  });
});
