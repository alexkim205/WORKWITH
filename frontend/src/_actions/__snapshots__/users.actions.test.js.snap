// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Actions it should DELETE a note 1`] = `
Array [
  Object {
    "type": "users/delete-user-pending",
  },
  Object {
    "error": null,
    "payload": undefined,
    "type": "users/delete-user-success",
  },
]
`;

exports[`User Actions it should GET a user 1`] = `
Array [
  Object {
    "type": "users/get-user-pending",
  },
  Object {
    "error": null,
    "payload": Object {
      "user": Object {
        "_id": "5e0bb5c61c90231c34596f83",
        "contacts": Array [
          "5e02efdd94748b21482e6010",
        ],
        "email": "alexkim@dev.com",
        "name": "Alex Kim",
        "role": "User",
      },
    },
    "type": "users/get-user-success",
  },
]
`;

exports[`User Actions it should GET users 1`] = `
Array [
  Object {
    "type": "users/get-users-pending",
  },
  Object {
    "error": null,
    "payload": Object {
      "users": Array [
        Object {
          "_id": "5e0bb5c61c90231c34596f83",
          "email": "alexkim@dev.com",
          "name": "Alex Kim",
          "role": "Admin",
        },
      ],
    },
    "type": "users/get-users-success",
  },
]
`;

exports[`User Actions it should LOGIN a user 1`] = `
Array [
  Object {
    "type": "users/login-pending",
  },
  Object {
    "error": null,
    "payload": Object {
      "refreshToken": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJfaWQiOiI1ZTBiYjVjNjFjOTAyMzFjMzQ1OTZmODMiLCJuYW1lIjoiQWxleCB0aGUgQm9vYiIsImVtYWlsIjoiYWxleGtpbUBkZXYuY29tIiwicm9sZSI6IlVTRVIiLCJpYXQiOjE1ODMxNzM5MjAsImV4cCI6MTU5MTQ2ODMyMH0.-c2fVoAVWTf-8JrinCIpL7U-jymRWUGW-c3_lIMogx4",
      "token": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE1Nzg1NDYxMjUsImlhdCI6MTU3Nzk0MTMyNX0.ksgtQ8uDQ3322tOhQ4jIkzZjH6B0luhLA1NRKGCG00w",
      "user": Object {
        "_id": "5e0bb5c61c90231c34596f83",
        "email": "alexkim@dev.com",
        "name": "Alex Kim",
        "role": "User",
      },
    },
    "type": "users/login-success",
  },
]
`;

exports[`User Actions it should LOGOUT a user 1`] = `
Array [
  Object {
    "type": "users/logout-success",
  },
]
`;

exports[`User Actions it should REGISTER a user 1`] = `
Array [
  Object {
    "type": "users/register-pending",
  },
  Object {
    "error": null,
    "payload": Object {
      "refreshToken": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJfaWQiOiI1ZTBiYjVjNjFjOTAyMzFjMzQ1OTZmODMiLCJuYW1lIjoiQWxleCB0aGUgQm9vYiIsImVtYWlsIjoiYWxleGtpbUBkZXYuY29tIiwicm9sZSI6IlVTRVIiLCJpYXQiOjE1ODMxNzM5MjAsImV4cCI6MTU5MTQ2ODMyMH0.-c2fVoAVWTf-8JrinCIpL7U-jymRWUGW-c3_lIMogx4",
      "token": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE1Nzg1NDYxMjUsImlhdCI6MTU3Nzk0MTMyNX0.ksgtQ8uDQ3322tOhQ4jIkzZjH6B0luhLA1NRKGCG00w",
      "user": Object {
        "_id": "5e0bb5c61c90231c34596f83",
        "email": "alexkim@dev.com",
        "name": "Alex Kim",
        "role": "User",
      },
    },
    "type": "users/register-success",
  },
]
`;

exports[`User Actions it should UPDATE a user 1`] = `
Array [
  Object {
    "type": "users/update-user-pending",
  },
  Object {
    "error": null,
    "payload": Object {
      "user": Object {
        "_id": "5e0bb5c61c90231c34596f83",
        "contacts": Array [
          "5e02efdd94748b21482e6010",
        ],
        "email": "alexkim@dev.com",
        "name": "Alex Kim",
        "role": "User",
      },
    },
    "type": "users/update-user-success",
  },
]
`;
